pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -6598,6 +6598,99 @@ spec:
 	}
 }
 
+func TestReconcile_ParamsValidationsImmediatelyFailPipelineRun(t *testing.T) {
+	names.TestingSeed()
+
+	ctx := context.Background()
+	cfg := config.NewStore(logtesting.TestLogger(t))
+	ctx = cfg.ToContext(ctx)
+
+	prs := []*v1beta1.PipelineRun{
+		parse.MustParseV1beta1PipelineRun(t, `
+metadata:
+  name: pipelinerun-matrix-param-invalid-type
+  namespace: foo
+spec:
+  pipelineSpec:
+   tasks:
+    - name: mytask
+      taskSpec:
+        steps:
+         - name: echo
+           image: alpine
+           script: |
+             echo "test"
+    - name: platforms-and-browsers
+      taskRef:
+        name: mytask
+      matrix:
+        params:
+          - name: platform
+            value: linux
+          - name: browser
+            value:
+              - chrome
+              - safari
+              - firefox
```

Comment:
is this relevant to the test case? If not, can it be deleted?

Reply:
Removed additional params